bootstrap/092_service_account_setup.sql

```diff
@@ -0,0 +1,29 @@
+
+CREATE OR REPLACE PROCEDURE admin.finalize_setup_from_service_account(token varchar, url varchar, web_url varchar)
+RETURNS varchar
+LANGUAGE sql
+as
+begin
+    execute immediate 'create or replace function internal.get_ef_url() returns string as \'\\\'' || url || '\\\'\';';
+    execute immediate 'create or replace function internal.get_tenant_url() returns string as \'\\\'' || web_url || '\\\'\';';
+
+    call internal.set_config('tenant_url', web_url);
+    call internal.set_config('url', url);
+    call admin.update_reference('OPSCENTER_API_INTEGRATION', 'ADD', SYSTEM$REFERENCE('API Integration', 'OPSCENTER_API_INTEGRATION', 'persistent', 'usage'));
+    call admin.connect_sundeck(token);
+    call admin.finalize_setup();
+end;
+
+CREATE OR REPLACE PROCEDURE admin.upgrade_check()
+returns varchar
+language sql
+as
+begin
+    let version varchar;
+    call internal.get_config('post_setup') into :version;
+    let setup_version varchar := (select internal.setup_version());
+    if (:version <> :setup_version) then
```

Comment:
It's possible that `:version` is `NULL` (user never opened native app). Should this be something like `if (:version is NULL or :version <> :setup_version)`?

Reply:
yup, fixed